go vet fails during make test

On Ubuntu 14.04.4 amd64 minimal netboot ISO installation

# Environment
GOPATH=/home/user/WKS/snap-build
GOROOT=/home/user/GO/go-1.6.1
PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/home/user/GO/go-1.6.1/bin:/home/user/WKS/snap-build/bin

# Got go binary archive from official web site and extracted it
$ mkdir -p ~/GO
$ cd ~/GO
$ wget https://storage.googleapis.com/golang/go1.6.1.linux-amd64.tar.gz
$ tar zxf go1.6.1.linux-amd64.tar.gz
$ mv go go-1.6.1
$ type go
go is hashed (/home/user/GO/go-1.6.1/bin/go)

$ go version
go version go1.6.1 linux/amd64

Ran the following commands:

$ go get github.com/intelsdi-x/snap
$ go get github.com/tools/godep
$ make

$ make test
export SNAP_PATH=`pwd`/build; bash -c "./scripts/test.sh"
Getting GoConvey if not found
Getting goimports if not found
Getting cover if not found
gofmt
goimports
go vet
control/config.go:82: github.com/vrischmann/jsonutil.Duration composite literal uses unkeyed fields
control/control_test.go:1013: github.com/vrischmann/jsonutil.Duration composite literal uses unkeyed fields
exit status 1
snapd.go:617: github.com/vrischmann/jsonutil.Duration composite literal uses unkeyed fields
exit status 1
make: *** [test] Error 1

Latest commit in /home/user/WKS/snap-build/src/github.com/intelsdi-x/snap master branch is:

commit 388e91da912792f3234c5d6d5c7710e5c2308244
Merge: 8f2a67a 2d02134
Author: Matt Brender <mjbrender@gmail.com>
Date:   Tue Apr 19 08:29:05 2016 -0700

    Merge pull request #859 from mjbrender/master
    
    Clearer README introduction, added image and blog links

--- a/scripts/test.sh
+++ b/scripts/test.sh
@@ -20,9 +20,9 @@
 # The script does automatic checking on a Go package and its sub-packages, including:
 # 1. gofmt         (http://golang.org/cmd/gofmt/)
 # 2. goimports     (https://github.com/bradfitz/goimports)
-# 3. golint        (https://github.com/golang/lint)
-# 4. go vet        (http://golang.org/cmd/vet)
-# 5. race detector (http://blog.golang.org/race-detector)
+# 3. golint        (https://github.com/golang/lint) (disabled)
+# 4. go vet        (http://golang.org/cmd/vet) (disabled)
+# 5. race detector (http://blog.golang.org/race-detector) (disabled)
 # 6. test coverage (http://blog.golang.org/cover)
 
 # If the following plugins don't exist, exit
@@ -32,7 +32,7 @@
 [ -f $SNAP_PATH/plugin/snap-publisher-file ] || { echo 'Error: $SNAP_PATH/plugin/snap-publisher-file does not exist. Run make to build it.' ; exit 1; }
 
 TEST_DIRS="cmd/ control/ core/ mgmt/ pkg/ snapd.go scheduler/"
-VET_DIRS="./cmd/... ./control/... ./core/... ./mgmt/... ./pkg/... ./scheduler/... ."
+# VET_DIRS="./cmd/... ./control/... ./core/... ./mgmt/... ./pkg/... ./scheduler/... ."
 
 set -e
 
@@ -61,8 +61,13 @@ test -z "$(goimports -l -d $TEST_DIRS | tee /dev/stderr)"
 # echo "golint"
 # golint ./...
 
-echo "go vet"
-go vet $VET_DIRS
+# Disabling running go vet currently due to the inconsistency in it's reported
+# outputs depending on whether the project was pulled down via 'go get' or 'git clone'.
+# We will look to re-enable go vet checking when we can provide consistency in outputs
+# no matter how a developer gets the project.
+#echo "go vet"
+#go vet $VET_DIRS
+
 # go test -race ./... - Lets disable for now
 
 # Run test coverage on each subdirectories and merge the coverage profile.

